app/controllers/interactions_controller.rb

```diff
@@ -0,0 +1,9 @@
+class InteractionsController < ApplicationController
+  def index
+    # We parameterized this label to make it look like a slug for
+    # the URL, so now we need to turn it back into a string so that
+    # the service can be found from it. There's probably a better way.
+    @service = Service.find_by_label(params[:service_label].downcase.gsub("-"," "))
```

Comment:
You're right that there's probably a better way, it's a bit more work though: add a slug field to services and generate it on create. It's unlikely that this reverse parameterization is going to do the right thing, parameterisation destroys information (e.g. "Pavements - personal injury".parameterize gives "pavements-personal-injury" but reversing it "pavements-personal-injury".downcase.gsub("-"," ") gives "pavements personal injury")

(resurrecting a comment left on a non-PR version of this that commit that has since been rebased away - apols if you got this twice)